### PR TITLE
Plasma node implementation

### PR DESCRIPTION
**Pull request by jdkanani**
Regarding #3

### CODE REVIEW

File: contracts/RootChain.sol [hunk 1]
```diff
@@ -109,6 +113,9 @@ contract RootChain {
       root: root,
       createdAt: block.timestamp
     });
+    ChildBlockCreated(currentChildBlock, root);
+    DepositBlockCreated(currentChildBlock, root, txBytes);
```
Comment (federicobond) [inline contracts/RootChain.sol:117]: This is a great way to ensure the deposit transaction is broadcasted. Not sure if we have to include the root again though.

Comment (jdkanani) [reply contracts/RootChain.sol:117]: I added it so I can verify (or use directly without calculating it again). We can remove it, no problem.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+// start app
+require('./server')
+
+// setup chain
+const chain = require('./chain').default
+chain.start()
+
+process.on('SIGINT', function() {
+  console.log('Stoping plasma chain')
```
Comment (federicobond) [inline src/index.js]: Stopping

**Comment by federicobond** [comment]
I think this looks pretty good. sendTx is missing, should I wait until that one is ready or merge as it is? The client code would need unit tests too.

**Comment by jdkanani** [comment]
Let's wait. Only added deposit part. Need to add sendTx, block creations & validations etc..

**Pull request merged**